pynio/block.py

```diff
@@ -40,3 +40,11 @@ def name(self, value):
     @property
     def type(self):
         return self._type
+
+    def delete(self):
+        '''Delete self from instance and services'''
+        self._instance.blocks.remove(self._name)
+        for s in self._instance.services.values():
+            s._remove(self)
```

Comment:
It looks like this is only removing the block from local configs. Is that necessary? If these modified services are never saved, then nio will still have references to this deleted block. I actually think it's ok to leave the references in the services. What if you are deleting a block named "my_block" and your plan is to create a new "my_block" of a different type but still have it used in all the services?